modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -67,12 +67,10 @@ def wait(*args, **kwargs):
             # serially kick off all the deferred computations so that they can
             # all run asynchronously, then wait on all the results.
             [part.drain_call_queue() for part in partitions.flatten()]
-            # need to go through all the values of the map iterator
-            # since `wait` does not return anything, we need to explicitly add
-            # the return `True` value from the lambda
-            # TODO(https://github.com/modin-project/modin/issues/4491): Wait
-            # for all the partitions in parallel.
-            all(map(lambda partition: partition.wait() or True, partitions.flatten()))
+            # The partition manager invokes the relevant .wait() method under
+            # the hood, which should wait in parallel for all computations to finish
+            partition_mgr_cls = args[0]
```

Comment:
if this is always the case, we should probably change the signature of `wait()` to declare it a classmethod, i.e.
```python
@wraps(func)
def wait(cls, *args, **kwargs):
    result = func(cls, *args, **kwargs)
    # do the rest of the thing
    partitions = partitions.flatten()
    [part.drain_call_queue() for part in partitions]
    cls.wait_partitions(partitions)
```

I also wonder if it is the `.wait_partitions()` method which should actually be draining the call queues...